OrbitGl/ThreadTrack.cpp

```diff
@@ -139,17 +144,27 @@ void ThreadTrack::Draw(GlCanvas* canvas, PickingMode picking_mode) {
   event_track_->SetPos(pos_[0], pos_[1]);
   event_track_->SetSize(canvas->GetWorldWidth(), event_track_height);
   event_track_->Draw(canvas, picking_mode);
+
+  tracepoint_track_->SetPos(pos_[0], pos_[1]);
+  tracepoint_track_->SetSize(canvas->GetWorldWidth(), event_track_height);
+  tracepoint_track_->Draw(canvas, picking_mode);
 }
 
 void ThreadTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick, PickingMode picking_mode) {
   event_track_->SetPos(pos_[0], pos_[1]);
   event_track_->UpdatePrimitives(min_tick, max_tick, picking_mode);
+
+  tracepoint_track_->SetPos(pos_[0], pos_[1]);
+  tracepoint_track_->UpdatePrimitives(min_tick, max_tick, picking_mode);
+
+  TimerTrack::SetTracepointTrack(tracepoint_track_);
```

Comment:
Thank you for your message. It is needed because in TimerTrack::GetYFromDepth we need to know if any tracepoints have been triggered to be able to reposition the timer track accordingly to suit the tracepoint track. This information belongs to  tracepoint_track_ and this is why I made it a member.

Reply:
I used a virtual function as you have suggested and removed having tracepoint_track_  as a member. Thank you!